stubs/regex/regex/_regex.pyi

```diff
@@ -0,0 +1,184 @@
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> Match[AnyStr] | None: ...
+    def split(
+        self, string: AnyStr, maxsplit: int = ..., concurrent: bool | None = ..., timeout: int | None = ...
+    ) -> list[AnyStr]: ...
+    def splititer(
+        self, string: AnyStr, maxsplit: int = ..., concurrent: bool | None = ..., timeout: int | None = ...
+    ) -> Splitter[AnyStr]: ...
+    def findall(
+        self,
+        string: AnyStr,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        overlapped: bool = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> list[Any]: ...
+    def finditer(
+        self,
+        string: AnyStr,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        overlapped: bool = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> Scanner[AnyStr]: ...
+    def sub(
+        self,
+        repl: AnyStr | Callable[[Match[AnyStr]], AnyStr],
+        string: AnyStr,
+        count: int = ...,
+        flags: int = ...,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> AnyStr: ...
+    def subf(
+        self,
+        format: AnyStr | Callable[[Match[AnyStr]], AnyStr],
+        string: AnyStr,
+        count: int = ...,
+        flags: int = ...,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> AnyStr: ...
+    def subn(
+        self,
+        repl: AnyStr | Callable[[Match[AnyStr]], AnyStr],
+        string: AnyStr,
+        count: int = ...,
+        flags: int = ...,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> tuple[AnyStr, int]: ...
+    def subfn(
+        self,
+        format: AnyStr | Callable[[Match[AnyStr]], AnyStr],
+        string: AnyStr,
+        count: int = ...,
+        flags: int = ...,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> tuple[AnyStr, int]: ...
+    def scanner(
+        self,
+        string: AnyStr,
+        pos: int | None = ...,
+        endpos: int | None = ...,
+        overlapped: bool = ...,
+        concurrent: bool | None = ...,
+        timeout: int | None = ...,
+    ) -> Scanner[AnyStr]: ...
+
+class Match(Generic[AnyStr]):
+
+    re: Pattern[AnyStr]
+    string: AnyStr
+    pos: int
+    endpos: int
+    partial: bool
+    regs: tuple[tuple[int, int], ...]
+    fuzzy_counts: tuple[int, int, int]
+    fuzzy_changes: tuple[list[int], list[int], list[int]]
+    lastgroup: str | None
+    lastindex: int | None
+    @overload
+    def group(self, __group: int | str = ...) -> AnyStr | None: ...
```

Comment:
This will be annoying in practice, because you need:
```python3
value = match.group(1)
assert value is not None  # This is annoying
something_that_expects_string(value)
```

instead of `something_that_expects_string(match.group(1))`. It would be best to handle this with `| Any`, as in `stdlib/re.pyi`. I recently explained this here: https://github.com/python/typing/discussions/994#discussioncomment-1871463

Reply:
Makes sense, thanks! I assume that applies to anything returning `| None`.
So, for example, I would also type the return of `regex.search` as `_regex.Match[AnyStr] | Any` instead of `_regex.Match[AnyStr] | None`?

Reply:
`| None` makes sense when the user should check whether the result is None. This makes sense for some methods. The annotations in `re.pyi` have been checked with this in mind, so you should be able to just do whatever it does.

Reply:
I've aligned the returns with `re.pyi` and likewise with `typing.pyi` for `Pattern`/`Match`/etc